src/mutex.rs

```diff
@@ -85,6 +85,113 @@ impl<T> Mutex<T> {
     }
 }
 
+macro_rules! acquire_slow_method {
+    ($name:ident, $listener:ident, $wait:expr, $doc:literal$(, $maybe_async:ident)?$(,)?) => {
+        #[doc = $doc]
+        #[cold]
+        $($maybe_async)? fn $name(&self) {
+            // Get the current time.
+            #[cfg(not(target_arch = "wasm32"))]
+            let start = Instant::now();
+
+            loop {
+                // Start listening for events.
+                let $listener = self.lock_ops.listen();
+
+                // Try locking if nobody is being starved.
+                match self
+                    .state
+                    .compare_exchange(0, 1, Ordering::Acquire, Ordering::Acquire)
+                    .unwrap_or_else(|x| x)
+                {
+                    // Lock acquired!
+                    0 => return,
+
+                    // Lock is held and nobody is starved.
+                    1 => {}
+
+                    // Somebody is starved.
+                    _ => break,
+                }
```

Comment:
Can we factor this block of code into a separate method? It is repeated 2 times in this method with small variations in the `_=>` case, and with this PR applied would then be duplicated again, resulting in 5 repetitions...

Reply:
I'm not sure we can factor that into a method, since it includes control flow. I could make it a macro if you like.

Reply:
> since it includes control flow

that can be factored out, using something like the [`core`](https://doc.rust-lang.org/core/index.html)::[`ops`](https://doc.rust-lang.org/core/ops/index.html)::[`ControlFlow`](https://doc.rust-lang.org/core/ops/enum.ControlFlow.html), e.g.:
```rust
enum ScxResult {
    /// Lock acquired
    Acquired,
    /// Lock is held and nobody is starved
    HeldUnstarved,
    /// Somebody is starved
    Starved,
}
fn do_scx(state: &...) -> ScxResult {
    use ScxResult::*;
    match state
        .compare_exchange(0, 1, Ordering::Acquire, Ordering::Acquire)
        .unwrap_or_else(|x| x)
    {
        0 => Acquired,
        1 => HeldUnstarved,
        _ => Starved,
    }
}
```

Reply:
Maybe we could refactor this into a trait, like I did for the `OnceCell` in #27.